src/valid/expression.rs

```diff
@@ -151,6 +153,33 @@ impl<'a> std::ops::Index<Handle<crate::Expression>> for ExpressionTypeResolver<'
 
 #[cfg(feature = "validate")]
 impl super::Validator {
+    pub(super) fn validate_const_expression(
+        &self,
+        handle: Handle<crate::Expression>,
+        resolve_context: &crate::proc::ResolveContext,
+        gctx: crate::proc::GlobalCtx,
+        mod_info: &mut ModuleInfo,
+    ) -> Result<(), super::ConstExpressionError> {
+        use crate::Expression as E;
+
+        match gctx.const_expressions[handle] {
+            E::Literal(_) | E::Constant(_) | E::ZeroValue(_) => {}
+            E::Compose { ref components, ty } => {
+                validate_compose(
+                    ty,
+                    gctx,
+                    components.iter().map(|&handle| mod_info[handle].clone()),
+                )?;
+            }
+            _ => return Err(super::ConstExpressionError::NonConst),
+        }
+
+        mod_info.const_expr_types[handle.index()] =
```

Comment:
Since this has been pulled out into `process_const_expression`, a clear analog to `process_expression`, I think I agree this doesn't need a comment any more.